File: back_end_fg/WebApplication1/WebApplication1/Models/PriceToWin.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WimsApiMKI.Models
{
    public class PriceToWin
    {
        private int _id;
        private WimsUser _twitcher;
        private DateTime _offerEnd;
        private Game _game;
        private int _auctionStartValue;
        private string _description;
        private bool _active;

        public PriceToWin(int value)
        {
            Id = value;
        }

        public PriceToWin()
        {
            Twitcher = new WimsUser();
            OfferEnd = new DateTime();
            Game = new Game();
        }

        #region GetSet
        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public WimsUser Twitcher
        {
            get { return _twitcher; }
            set { _twitcher = value; }
        }

        public DateTime OfferEnd
        {
            get { return _offerEnd; }
            set { _offerEnd = value; }
        }

        public Game Game
        {
            get { return _game; }
            set { _game = value; }
        }

        public int AuctionStartValue
        {
            get { return _auctionStartValue; }
            set { _auctionStartValue = value; }
        }

        public string Description
        {
            get { return _description; }
            set { _description = value; }
        }

        public bool Active
        {
            get { return _active; }
            set { _active = value; }
        }
        #endregion
    }
}

File: WIMS-front/src/app/Observables/PriceToWin.ts
import {Game} from './Game';
import {Twitcher} from './Twitcher';

export class PriceToWin {

    private id: number;
    private twitcher: Twitcher;
    private active: boolean;
    private game: Game;
    constructor(

        private offerEnd: string,

        // private _GameId: number,
        private auctionStartValue: number,
        private description: string,

    ) {}
    private restantTotal: any;
    get RestantTotal(): any {
        return this.restantTotal;
    }

    set RestantTotal(value: any) {
        this.restantTotal = value;
    }
    get Id(): number {
        return this.id;
    }

    set Id(value: number) {
        this.id = value;
    }

    get Twitcher(): Twitcher {
        return this.twitcher;
    }

    set Twitcher(value: Twitcher) {
        this.twitcher = value;
    }

    get OfferEnd(): string {
        return this.offerEnd;
    }

    set OfferEnd(value: string) {
        this.offerEnd = value;
    }

    get Game(): Game {
        return this.game;
    }

    set Game(value: Game) {
        this.game = value;
    }

    // get GameId(): number {
    //     return this._GameId;
    // }
    //
    // set GameId(value: number) {
    //     this._GameId = value;
    // }

    get AuctionStartValue(): number {
        return this.auctionStartValue;
    }

    set AuctionStartValue(value: number) {
        this.auctionStartValue = value;
    }

    get Description(): string {
        return this.description;
    }

    set Description(value: string) {
        this.description = value;
    }

    get Active(): boolean {
        return this.active;
    }

    set Active(value: boolean) {
        this.active = value;
    }
}
